<launch>
  <arg name="namespace" default="taluy" />
  <arg name="name" default="rtsp_camera" />
  <arg name="compressed" default="true" />
  <arg name="verbose" default="true" />
  <arg name="pub_camera_info" default="false" />
  <!-- RTSP Parameters -->
  <arg name="rtsp_url" default="rtsp://192.168.1.65/554" />
  <arg name="latency" default="0" />
  <arg name="buffer_mode" default="auto" />
  <arg name="drop_on_latency" default="true" />
  
  <!-- Reconnection Parameters -->
  <arg name="reconnect_interval_sec" default="5" />
  <arg name="max_consecutive_failures" default="100" />
  
  <!-- Video Parameters -->
  <arg name="width" default="1920" />
  <arg name="height" default="1080" />
  <arg name="fps" default="30" />

  <group ns="$(arg namespace)/cameras">
    <group ns="$(arg name)/image_raw">
      <!-- Disable unused plugins -->
      <group if="$(arg compressed)">
        <rosparam param="disable_pub_plugins">["image_transport/compressedDepth",
          "image_transport/theora"]</rosparam>
      </group>

      <!-- Disable unused plugins and compressed -->
      <group unless="$(arg compressed)">
        <rosparam param="disable_pub_plugins">["image_transport/compressed",
          "image_transport/compressedDepth", "image_transport/theora"]</rosparam>
      </group>
    </group>

    <!-- Launch RTSP Camera -->
    <node name="$(arg name)" type="rtsp_cam_node" pkg="auv_cam" output="screen">
      <param name="verbose" type="bool" value="$(arg verbose)" />
      <param name="pub_camera_info" value="$(arg pub_camera_info)" />
      
      <!-- RTSP Parameters -->
      <param name="rtsp_url" value="$(arg rtsp_url)" />
      <param name="latency" value="$(arg latency)" />
      <param name="buffer_mode" value="$(arg buffer_mode)" />
      <param name="drop_on_latency" value="$(arg drop_on_latency)" />
      
      <!-- Reconnection Parameters -->
      <param name="reconnect_interval_sec" value="$(arg reconnect_interval_sec)" />
      <param name="max_consecutive_failures" value="$(arg max_consecutive_failures)" />
      
      <!-- Video Parameters -->
      <param name="width" value="$(arg width)" />
      <param name="height" value="$(arg height)" />
      <param name="fps" value="$(arg fps)" />
    </node>
  </group>

  <include file="$(find auv_cam)/launch/inc/image_proc.launch.xml">
    <arg name="camera_namespace" value="$(arg namespace)/cameras/$(arg name)" />
  </include>
</launch>
